tests/cpp/test_multidevice_tutorial.cpp

```diff
@@ -0,0 +1,379 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+
+#include <multidevice/communicator.h>
+#include <ops/all_ops.h>
+#include <tests/cpp/multidevice.h>
+
+namespace nvfuser {
+
+class MultiDeviceTutorial : public MultiDeviceTest {
+ protected:
+  static void SetUpTestSuite() {
+    verbose_ = getNvFuserEnv("TUTORIAL_VERBOSE");
+  }
+
+  void SetUp() {
+    MultiDeviceTest::SetUp();
+    if (communicator_->is_available() == false) {
+      GTEST_SKIP()
+          << "Distributed setting not available. "
+          << "Make sure you are on a node with n>1 GPUs and run "
+          << "`mpirun -np n -x TUTORIAL_VERBOSE=1 tutorial_multidevice`";
+    }
+  }
+
+ protected:
+  static bool verbose_;
+};
+
+bool MultiDeviceTutorial::verbose_ = false;
+
+// To run those tests, allocate a node with n>1 GPUs and run:
+//
+// mpirun -np n -x TUTORIAL_VERBOSE=1 tutorial_multidevice
+//
+// We use a SPMD paradigm, where each host process manages one and only device.
```

Comment:
Oh right! But we still are in a SPMD model right? IIUC, one the one hand we are SPMD, and on the other hand one-process = one device, right?

What would you write as a comment here?

Reply:
Yes! we are SPMD because each device executes the same program! I would just update to something like: We use a SPMD paradigm, where each host process manages one and only device and each device executes the same program.